src/vos/vos_dtx_iter.c

```diff
@@ -246,6 +246,15 @@ dtx_iter_fetch(struct vos_iterator *iter, vos_iter_entry_t *it_entry,
 	D_ASSERT(!dae->dae_aborted);
 	D_ASSERT(dae->dae_dth == NULL);
 
+	/*
+	 * The DTX entry is returned via the iteration. The upper layer caller, for DTX resync
+	 * or DTX cleanup, may abort it sometime later. Before being aborted, it may re-attach
+	 * to some DTX handle for handling resent request. Then there is race between them. So
+	 * before become committable or being committed by RPC handler, needs to check whether
+	 * it has been aborted or not.
+	 */
```

Comment:
It seems to me that we could always check if the dae is aborted or not to avoid such race? Is checking dae state a heavy operation so that we introduce this 'need_validation' flag to avoid dae state checking for most cases?

Reply:
The check needs to lookup the DTX entry with its ID in the active table (or the committed table), although the table is indexed, consider the table size, it is still some overhead, especially this is for every modification.